### PR TITLE
[4.0] Hidden-mail custom element improvements

### PR DESCRIPTION
**Pull request by C-Lodder**
### Summary of Changes

Improvements to the Hidden-mail custrom element

- Cache some values in the constructor
- ESLint fixes
- Use static method where applicable

### Testing Instructions

1. Create a custom module
2. Enable the `Prepare Content` option
3. In the editor, type in a couple of email addresses

### Expected result

Exactly the same result as before the patch

### To-do

- Need to look further into how Custom Elements are actually made async using a mutation observer, but that can be done in a separate PR

### CODE REVIEW

File: build/media/webcomponents/js/hidden-mail/hidden-mail.js [hunk 1]
```diff
@@ -1,50 +1,58 @@
...
+
+        if (nodeName) {
+          // We do care for some attributes
+          if (['is-link', 'is-email', 'first', 'last', 'text'].indexOf(nodeName) === -1) {
+            const { nodeValue } = this.attributes.item(i);
+
+            this.newElement.setAttribute(nodeName, nodeValue);
+          }
+        }
+      }
```
Comment (dgrammatiko) [inline build/media/webcomponents/js/hidden-mail/hidden-mail.js]: Can you swap the for with:
```js
          // Get all of the original element attributes, and pass them to the link
          [].slice.call(this.attributes).forEach((attribute) => {
            // We do care for some attributes
            if (['is-link', 'is-email', 'first', 'last', 'text'].indexOf(nodeName) === -1) {
              this.newElement.setAttribute(nodeName, attribute);
            }
          })
```

Comment (C-Lodder) [reply build/media/webcomponents/js/hidden-mail/hidden-mail.js]: @dgrammatiko done and also added the `disconnectedCallback`

Comment (dgrammatiko) [reply build/media/webcomponents/js/hidden-mail/hidden-mail.js]: I think it needs to be innerHTML as you want to clean up everything, innerText will keep any existing elements

Comment (C-Lodder) [reply build/media/webcomponents/js/hidden-mail/hidden-mail.js]: done

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**